File: TEO/Mav.Services/Abstract/ISectionService.cs
using Mav.Common.Models;
using Mav.Data.Repositories;
using Mav.Models.SearchModels;
using Mav.Models.ViewModels;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace Mav.Services.Abstract
{
    public interface ISectionService
    {
        Task<T> GetSectionById<T>(int id);

        Task<IEnumerable<T>> GetSectionList<T>(int assessmentId);

        Task<IPaginate<T>> GetSectionListPages<T>(SectionSearchModel vm);

        Task<GenericResult<T>> CreateSection<T>(AddSectionViewModel m);

        Task<GenericResult<T>> UpdateSection<T>(UpdateSectionViewModel m);

        Task<GenericResult<T>> DeleteSection<T>(int id);

        Task<int> GetSectionCount(int assessmentId);

    }
}

File: TEO/Mav.UI/Mav.Ui/src/app/services/i-section.service.ts
import { Observable } from 'rxjs/Observable';
import { GenericResult } from '../models/generic.result.model';
import { SectionListingModel } from '../models/search/section.listing.model';
import { AddSectionViewModel } from '../models/admin/add-section.view.model';
import { UpdateSectionViewModel } from '../models/admin/update-section.view.model';
import { SectionSearchModel } from '../models/search/section.search.model';
import { Paginate } from '../models/paginated.items';

export interface ISectionService {

    getSectionById<T>(id: number): Observable<T>;

    getSectionList(assessmentId: number): Observable<SectionListingModel[]>;

    getSectionListPages(filter: SectionSearchModel): Observable<Paginate<SectionListingModel>>;

    createSection(newItem: AddSectionViewModel): Observable<GenericResult<number>>;

    updateSection(item: UpdateSectionViewModel): Observable<GenericResult<number>>;

    deleteSection(id: number): Observable<GenericResult<number>>;

}
